graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/OperationModelBuilderBase.java

```diff
@@ -90,4 +94,66 @@ protected void validateNoFiltering(List<ConditionModel> whereConditions, EntityM
       throw SkipException.INSTANCE;
     }
   }
+
+  protected Optional<String> findFieldNameWithDirective(
+      String directiveName, GraphQLScalarType expectedType) throws SkipException {
+    return findFieldNameWithDirective(directiveName, Collections.singletonList(expectedType));
+  }
+
+  protected Optional<String> findFieldNameWithDirective(
+      String directiveName, List<GraphQLScalarType> expectedTypes) throws SkipException {
+    Optional<String> result = Optional.empty();
+    for (InputValueDefinition inputValue : operation.getInputValueDefinitions()) {
+      if (isDirective(inputValue, directiveName, expectedTypes)) {
+        if (result.isPresent()) {
+          invalidMapping(
+              "Query %s: @%s can be used on at most one argument (found %s and %s)",
+              operationName, directiveName, result.get(), inputValue.getName());
+          throw SkipException.INSTANCE;
+        }
+        result = Optional.of(inputValue.getName());
+      }
+    }
+    return result;
+  }
+
+  private boolean isDirective(
+      InputValueDefinition inputValue, String directiveName, List<GraphQLScalarType> expectedTypes)
+      throws SkipException {
+    boolean hasDirective = DirectiveHelper.getDirective(directiveName, inputValue).isPresent();
+    if (!hasDirective) {
+      return false;
+    }
+    Type<?> type = TypeHelper.unwrapNonNull(inputValue.getType());
+    boolean typeMatchFound = false;
+    for (GraphQLScalarType expectedType : expectedTypes) {
+      if (type instanceof TypeName && ((TypeName) type).getName().equals(expectedType.getName())) {
```

Comment:
You can use `TypeHelper.deepEquals` to shorten this.

Reply:
As far as I can see, the `expectedType` works only on the `Type` class. However, here I need to compare the `GraphQLScalarType`. I don't see a clear way to extract the `Type` out of `GraphQLScalarType`. Is it possible?

Reply:
You're right, I was mixing different things: `GraphQLScalarType` is not a subtype of `Type`.